### ISSUE TITLE
Stack overflow with circular reference in build and publishing

### ISSUE THREAD

**Issue by markfinal** [markfinal/BuildAMation#344]
Suppose you build DLL A, and application B that links against A.

Now publish, but use B as the anchor/reference, and A is dependent on it.

Seems to give the stack overflow:

```
Analysing module dependencies
Stack overflow in unmanaged: IP: 0x35022c, fault addr: 0xbf751ffc

Unhandled Exception:
System.OutOfMemoryException: Out of memory
  at (wrapper alloc) System.Object:AllocSmall (intptr,intptr)
  at Bam.Program.Main (System.String[] args) [0x001f4] in <ff5b7d4155ad465eb5dac5593e6bf948>:0
```

**Comment by markfinal** [comment]
Was this the same stack overflow that was fixed in #351?

**Comment by markfinal** [comment]
The description is also normal behaviour for publishing, so this isn't a very good reproducible case. I am closing it for that reason.
